src/main/java/dev/nolij/zson/Zson.java

```diff
@@ -127,6 +146,85 @@ public static String escape(String string, char escapeQuotes) {
 		return result.toString();
 	}
 
+	@NotNull
+	@Contract("_ -> new")
+	public static Map<String, ZsonValue> obj2Map(@NotNull Object object) {
+		Map<String, ZsonValue> map = Zson.object();
+		for (Field field : object.getClass().getDeclaredFields()) {
+			if(!shouldInclude(field, true)) continue;
+			Comment comment = field.getAnnotation(Comment.class);
+			String commentValue = comment == null ? null : comment.value();
+			try {
+				boolean accessible = field.canAccess(object);
+				if (!accessible) field.setAccessible(true);
+				map.put(field.getName(), new ZsonValue(commentValue, field.get(object)));
+				if (!accessible) field.setAccessible(false);
+			} catch (IllegalAccessException e) {
+				throw new RuntimeException("Failed to get field " + field.getName(), e);
+			}
+		}
+		return map;
+	}
+
+	@NotNull
+	@Contract("_ , _ -> new")
+	public static <T> T map2Obj(@NotNull Map<String, ZsonValue> map, @NotNull Class<T> type) {
+		try {
+			T object = type.getDeclaredConstructor().newInstance();
+			for (Field field : type.getDeclaredFields()) {
+				if(!shouldInclude(field, false)) continue;
+
+				if (!map.containsKey(field.getName())) {
+					throw new IllegalArgumentException("Missing field " + field.getName() + " in map");
+				}
+
+				boolean accessible = field.canAccess(object);
+				if (!accessible) field.setAccessible(true);
+				setField(field, object, map.get(field.getName()).value);
+				if (!accessible) field.setAccessible(false);
+			}
+			return object;
+		} catch (Exception e) {
+			throw new RuntimeException("Failed to create object of type " + type.getSimpleName(), e);
+		}
+	}
+
+	private static boolean shouldInclude(Field field, boolean includeStatic) {
+		Exclude exclude = field.getAnnotation(Exclude.class);
+		if (exclude != null) return false;
+		int modifiers = field.getModifiers();
+		if(Modifier.isTransient(modifiers)) return false;
+
+		Include include = field.getAnnotation(Include.class);
+		return (includeStatic || !Modifier.isStatic(modifiers)) && (include != null || Modifier.isPublic(modifiers));
+	}
+
+	private static <T> void setField(Field field, Object object, Object value) {
+		@SuppressWarnings("unchecked")
+		Class<T> type = (Class<T>) field.getType();
+		try {
+			if(type.isPrimitive()) {
+				switch (type.getName()) {
```

Comment:
I'm pretty sure Java supports something like the following:
```java
switch (x) {
    case int intX -> // now I have `int intX = (int) x;`
    case float floatX -> // now I have `float floatX = (float) x;`
    ...
}
```

Please switch to that if I'm not mistaken.

Reply:
That doesn't work, since we need to check the type of the field we are setting to, not the type of the `value` we are setting it to. `value` could be an `int` when the field is a `short`, so we need to know when to cast it to a `short`.